test/new-e2e/tests/containers/dump_cluster_state.go

```diff
@@ -286,4 +289,55 @@ func dumpK8sClusterState(ctx context.Context, kubeconfig *clientcmdapi.Config, o
 		fmt.Fprintf(out, "Failed to execute Get command: %v\n", err)
 		return
 	}
+
+	// Get the logs of containers that have restarted
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigFile.Name())
+	if err != nil {
+		fmt.Fprintf(out, "Failed to build Kubernetes config: %v\n", err)
+		return
+	}
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Fprintf(out, "Failed to create Kubernetes client: %v\n", err)
+		return
+	}
+
+	continueToken := "start"
+	for continueToken != "" {
+		if continueToken == "start" {
+			continueToken = ""
+		}
+		pods, err := k8sClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+			Continue: continueToken,
+		})
+		if err != nil {
+			fmt.Fprintf(out, "Failed to list pods: %v\n", err)
+			return
+		}
+		continueToken = pods.Continue
+
+		for _, pod := range pods.Items {
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.RestartCount > 0 {
+					fmt.Fprintf(out, "\nLOGS FOR POD %s/%s CONTAINER %s:\n", pod.Namespace, pod.Name, containerStatus.Name)
+					logs, err := k8sClient.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
+						Container: containerStatus.Name,
+						Previous:  true,
+						TailLines: pointer.Ptr(int64(20)),
```

Comment:
Why not all? We don't expect loads of logs

Reply:
I feared spamming the GitLab console if a verbose container (an agent with DEBUG log level or an nginx logging every single query) was OOM killed after 20 minutes.
But I can put a higher reasonable limit.